From: Jeff Mahoney <jeffm@suse.com>
Subject: mm: use addr instead of start to adjust guard page
Patch-mainline: Never, fixed as a side effect of
       259756fcc66e2d6dd732c0b4ff99f825e69b3ba2

 Upstream commit d7824370e26325c881b665350ce64fb0a4fde24a fixed up some
 user-visible aspects of the guard page handling, but accidentally
 introduced a bug where the number of pages faulted in would be correct
 but the guard page would be among them.

 This is due to 'start' not being used except as a check and 'addr' being
 used as the actual address to start from.

 Commit 259756fcc66e2d6dd732c0b4ff99f825e69b3ba2, which changed how the
 check worked by using a doubly-linked list, fixed this as a side-effect.

 This patch plucks out the important bit and advances 'addr' instead of
 'start'.

Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 mm/mlock.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/mm/mlock.c
+++ b/mm/mlock.c
@@ -173,7 +173,7 @@ static long __mlock_vma_pages_range(stru
 	/* We don't try to access the guard page of a stack vma */
 	if (vma->vm_flags & VM_GROWSDOWN) {
 		if (start == vma->vm_start) {
-			start += PAGE_SIZE;
+			addr += PAGE_SIZE;
 			nr_pages--;
 		}
 	}
